# Impact of Prompting Strategies on Information Extraction from AI News Articles Using Open-Source LLMs

In [3]:
import pandas as pd

# Import dataset
df = pd.read_csv("MIT_AI_ARTICLES.csv")

print("Original article number:", len(df))

# Hold on only necessary columns
df = df[[
    "title",
    "publication_date",
    "summary",
    "body",
    "url"
]]

# Remove the ones with an empty Body.
df = df.dropna(subset=["body"])

# Calculate word count
df["word_count"] = df["body"].astype(str).str.split().str.len()

# Remove shorter or longer articles
clean_df = df[
    (df["word_count"] >= 200) &
    (df["word_count"] <= 3000)
]

print("Number of cleaned articles:", len(clean_df))

# Kaydet
clean_df.to_csv("clean_articles.csv", index=False)

clean_df.head()

Original article number: 314
Number of cleaned articles: 314


,title,publication_date,summary,body,url,word_count
0,A new model predicts how molecules will dissol...,"August 19, 2025",Solubility predictions could make it easier to...,"Using machine learning, MIT chemical engineers...",https://news.mit.edu/2025/new-model-predicts-h...,1149
1,Researchers glimpse the inner workings of prot...,"August 18, 2025",A new approach can reveal the features AI mode...,"Within the past few years, models that can pre...",https://news.mit.edu/2025/researchers-glimpse-...,1025
2,How AI could speed the development of RNA vacc...,"August 15, 2025",MIT engineers used a machine-learning model to...,"Using artificial intelligence, MIT researchers...",https://news.mit.edu/2025/how-ai-could-speed-d...,1006
3,"Using generative AI, researchers design compou...","August 14, 2025",The team used two different AI approaches to d...,"With help from artificial intelligence, MIT re...",https://news.mit.edu/2025/using-generative-ai-...,1076
4,A new way to test how well AI systems classify...,"August 13, 2025",As large language models increasingly dominate...,Is this movie review a rave or a pan? Is this ...,https://news.mit.edu/2025/new-way-test-how-wel...,1186


In [13]:
clean_df["word_count"].describe()

count     314.000000
mean     1053.404459
std       283.800000
min       259.000000
25%       910.750000
50%      1051.500000
75%      1172.000000
max      2304.000000
Name: word_count, dtype: float64

After removing missing values and filtering articles outside the 200–3000 word range, all 314 articles remained in the dataset. 
Therefore, no additional article removal was required.

In [21]:
# Select 100 articles
sample_df = clean_df.sample(
    n=100,
    random_state=42
)

sample_df.to_csv(
    "evaluation_articles.csv",
    index=False
)

print("Selected articles:", len(sample_df))

Selected articles: 100
